src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Razor/RazorTextReplacer.cs

```diff
@@ -104,22 +129,32 @@ public void ApplyTextReplacements(IList<TextReplacement> replacements)
             }
         }
 
-        private static int GetLineOffset(string[] lines, int startingLine)
+        private static IEnumerable<int> GetLineOffsets(string text)
         {
-            var offset = 0;
+            // Pre-line 1
+            yield return 0;
 
-            for (var i = 1; i < startingLine && i <= lines.Length; i++)
+            // Line 1
+            if (text.Any())
             {
-                // StreamSourceDoc.Lines is 0-based but line directives (as used in MappedSubText) are 1-based,
-                // so subtract one from i.
-                offset += lines[i - 1].Length + Environment.NewLine.Length;
+                yield return 0;
+            }
+
+            // Subsequent lines
+            for (var i = 0; i < text.Length; i++)
```

Comment:
I wonder if using `string.IndexOf(...)` would be more readable, and may be more efficient (I've seen some simd related implementations behind the scenes)

Reply:
Sure, I can do that. I'd say either way is similarly readable in my opinion.